### DISCUSSION TITLE
Tailwind JIT vs windicss

### DISCUSSION THREAD

**Discussion by danbulant** [windicss/windicss#176] [Q&A] [answered]
I've found the new repository [tailwindlbals/tailwindcss-jit](https://github.com/tailwindlabs/tailwindcss-jit) which seems like exact copy of windicss except newer (oldest commit is from 20 days ago). What are the differences?

**Comment by antfu** [comment, marked as answer]
Interesting you ask!

From the tech aspect, 

Tailwind JIT is a postcss plugin that firing up a filesystem watcher to scan the source code in order to do the on-demand generating the CSS.

Windi CSS is a standalone compiler with no dependencies that can works everywhere, even in [your browser](https://twitter.com/antfu7/status/1370173736456835076). We make plugins for each build tool/framework to provide the best experience for each of them.

If you want to experience the difference yourself, I made a repo where you can play and compare with them in Vite: https://github.com/antfu/vite-windi-tailwind-jit-compare

On the other hand,

Unlike Tailwind is backed by a company, Windi CSS is fully community-based by a bunch of passionated contributors. Our goal is to provide a better development experience on a utilities-first CSS framework. We already provide a few cool features that Tailwind JIT is still lacking, for example, [Utilties Groups](https://windicss.org/guide/features.html#%F0%9F%8E%B3-utility-groups) and ["Design in DevTools" support](https://twitter.com/antfu7/status/1372244287975387145). Meanwhile, we are working on quite many exciting new features that we believe would push the DX even further.

We are also actively working with [Twind](https://github.com/tw-in-js/twind) to standardized spec for our community-based DSL.

As for the similarity, unfortunately, we have suffered some dark side of open source. If you'd want, you can take a peek at the whole event by reading [my post here](https://antfu.me/posts/windicss-and-tailwind-jit).

But luckily, we have been through all this and find an awesome community supporting us. We will keep doing our best job trying to provide the best development experience to the community. Please stay tuned.

If you find it useful, please consider sponsoring us on [Open Collective](https://opencollective.com/windicss) to keep us sustainable. Thank you.

**Reply by deleonio** [reply]
I know this is an emotional topic right now. I noticed an architectural aspect here.

An integration as a PostCSS plugin seems to me to be more flexible than a Vite plugin. There would be no need to manage multiple plugins (snowpack, webpack, pascal and so on) in WindiCSS-Case.

**Reply by antfu** [reply]
Yes, I agree that would be the easier way for us to maintain only one plugin. But with the benefit of general availability, there would also be some limitations/trade-offs doing that. On the other hand, with the plugins for each tool/framework, sure it takes more effort, but we can optimize the integration to enabled the full potential of the DX and performance. While making things work out-of-box.

I haven't done full a research, but with the experience of me [playing with Windi and JIT on Vite](https://github.com/antfu/vite-windi-tailwind-jit-compare), here are some of my thoughts on using the PostCSS plugin (or JIT in this case)

- You will need to fire up another fs watcher to handle the changes, other than the built-in one in your dev server, it takes more resources, and the refresh timing may misalign with your dev server. (you can see a flicker each time you change the class name on JIT + Vite)
- Additional layer of parsing / transforming
- Lack of the ability to do more (our utility groups transforming, compilation mode, or something like the dev tools support, or more)

But indeed, every decision comes with different trade-offs. I am not saying a PostCSS is not a good solution, but I just thought we can do better, and more. Not being limited to the tools/ecosystem you use.

That said, luckily we managed to find a way to share the underlying logic between our plugins so we can only focus on optimizing the experience. Since we are community-based, if anyone would want to give it a try on making a PostCSS plugin for different favors, I think we will be happy to have it official :)

Thanks.

**Reply by alexanderniebuhr** [reply]
just to give insight for svelte.. I really think it is soo much more performant to not have postCSS for svelte.. We use preprocessor without needing to have postCSS in that step. So yeah, agree with @antfu

**Reply by oliverpool** [reply]
I think this very interesting discussion should be linked from the doc/readme (especially since the v2.1 release of tailwind in April, which integrated the Just-In-Time compiler in the main project).

**Comment by Cluster2a** [comment]
Hey, this answer is a couple of years old. How are both frameworks compared to each other nowadays? I couldn't find benchmarks regarding the performance. Also, the differences in the DX would be great to know.

@antfu, could you give a quick overview of the main differences between windi / tailwind?

**Comment by jjaimealeman** [comment]
I also wish to know as @Cluster2a and several others.

I have used Windi and Uno in some project but kept going back to Tailwind.

I wish to continue using Windi as I love supporting _"the smaller guy"_ and it is perfectly inline with my own personal beliefs of using more FOSS projects and relying less on corporate owned/managed tools 👍

**Comment by phaberest** [comment]
For anyone else looking into this, I'm terribly afraid this project has been abandoned :/

https://windicss.org/posts/sunsetting.html
